python/cudf/cudf/core/frame.py

```diff
@@ -35,31 +35,56 @@ class Frame(libcudf.table.Table):
     def _from_table(cls, table):
         return cls(table._data, index=table._index)
 
+    @property
+    def empty(self):
+        """
+        Indicator whether Frame is empty.
```

Comment:
Use `cuDF object` or `Series/DataFrame/Index` instead of `Frame` here so the docstring is generic, and remove all the `empty` methods of those classes.

Reply:
I don't think we expose `Frame` through our docs and the inherited methods don't show up in the sphinx currently.

Reply:
We can enable inherited API docs: https://stackoverflow.com/questions/17254854/sphinx-autodoc-show-inheritance-how-to-skip-undocumented-intermediate-bases

But will have to keep disabling intermediate Bases for all the classes.

Reply:
Hmm -- does this work?

https://stackoverflow.com/questions/30344289/how-can-i-show-inherited-members-of-a-class-in-my-sphinx-documentation

Reply:
Having subclasses just call `super().method()` seems smelly. Even worse, we need to keep 4(!) copies of this docstring in sync every time a change is made.

Reply:
> Having subclasses just call `super().method()` seems smelly. Even worse, we need to keep 4(!) copies of this docstring in sync every time a change is made.

Agree



> Hmm -- does this work?
> 
> https://stackoverflow.com/questions/30344289/how-can-i-show-inherited-members-of-a-class-in-my-sphinx-documentation

I'll give it a try as that functionality exists in sphnix and probably make a similar change to a few other APIs that are doing the same.

Reply:
> But will have to keep disabling intermediate Bases for all the classes.

I don't understand -- which intermediate bases?

Reply:
How about we tackle the sphinx changes in a follow up PR as it seems orthogonal to this PR? For this PR, I'd suggest just having a single `Frame.empty()` method.

Reply:
> I don't understand -- which intermediate bases?

It was an example where a class inherited from two classes.



> How about we tackle the sphinx changes in a follow up PR as it seems orthogonal to this PR? For this PR, I'd suggest just having a single `Frame.empty()` method.

Sure 👍